drivers/spi/spi_andes_atcspi200.c

```diff
@@ -0,0 +1,629 @@
+#define REG_INTST_END_INT_OFFSET (4)
+#define REG_INTST_SLV_CMD_INT_OFFSET (5)
+
+#define REG_INTST_RX_FIFO_OR_INT_MSK (0x1 << REG_INTST_RX_FIFO_OR_INT_OFFSET)
+#define REG_INTST_TX_FIFO_UR_INT_MSK (0x1 << REG_INTST_TX_FIFO_UR_INT_OFFSET)
+#define REG_INTST_RX_FIFO_INT_MSK (0x1 << REG_INTST_RX_FIFO_INT_OFFSET)
+#define REG_INTST_TX_FIFO_INT_MSK (0x1 << REG_INTST_TX_FIFO_INT_OFFSET)
+#define REG_INTST_END_INT_MSK (0x1 << REG_INTST_END_INT_OFFSET)
+#define REG_INTST_SLV_CMD_INT_MSK (0x1 << REG_INTST_SLV_CMD_INT_OFFSET)
+
+/* Field mask of SPI config register */
+#define REG_CONFIG_RX_FIFO_SIZE_OFFSET (0)
+#define REG_CONFIG_TX_FIFO_SIZE_OFFSET (4)
+#define REG_CONFIG_DUAL_SPI_OFFSET (8)
+#define REG_CONFIG_QUAD_SPI_OFFSET (9)
+#define REG_CONFIG_DIRECT_IO_OFFSET (11)
+#define REG_CONFIG_AHB_MEM_OFFSET (12)
+#define REG_CONFIG_EILM_MEM_OFFSET (13)
+#define REG_CONFIG_SLV_OFFSET (14)
+
+#define REG_CONFIG_RX_FIFO_SIZE_MSK (0x3 << REG_CONFIG_RX_FIFO_SIZE_OFFSET)
+#define REG_CONFIG_TX_FIFO_SIZE_MSK (0x3 << REG_CONFIG_TX_FIFO_SIZE_OFFSET)
+#define REG_CONFIG_DUAL_SPI_MSK (0x1 << REG_CONFIG_DUAL_SPI_OFFSET)
+#define REG_CONFIG_QUAD_SPI_MSK (0x1 << REG_CONFIG_QUAD_SPI_OFFSET)
+#define REG_CONFIG_DIRECT_IO_MSK (0x1 << REG_CONFIG_DIRECT_IO_OFFSET)
+#define REG_CONFIG_AHB_MEM_MSK (0x1 << REG_CONFIG_AHB_MEM_OFFSET)
+#define REG_CONFIG_EILM_MEM_MSK (0x1 << REG_CONFIG_EILM_MEM_OFFSET)
+#define REG_CONFIG_SLV_MSK (0x1 << REG_CONFIG_SLV_OFFSET)
+
+/* Field mask of SPI control register */
+#define REG_CTRL_SPI_RST_OFFSET (0)
+#define REG_CTRL_RX_FIFO_RST_OFFSET (1)
+#define REG_CTRL_TX_FIFO_RST_OFFSET (2)
+#define REG_CTRL_RX_DMA_EN_OFFSET (3)
+#define REG_CTRL_TX_DMA_EN_OFFSET (4)
+#define REG_CTRL_RX_THRES_OFFSET (8)
+#define REG_CTRL_TX_THRES_OFFSET (16)
+
+#define REG_CTRL_SPI_RST_MSK (0x1 << REG_CTRL_SPI_RST_OFFSET)
+#define REG_CTRL_RX_FIFO_RST_MSK (0x1 << REG_CTRL_RX_FIFO_RST_OFFSET)
+#define REG_CTRL_TX_FIFO_RST_MSK (0x1 << REG_CTRL_TX_FIFO_RST_OFFSET)
+#define REG_CTRL_RX_DMA_EN_MSK (0x1 << REG_CTRL_RX_DMA_EN_OFFSET)
+#define REG_CTRL_TX_DMA_EN_MSK (0x1 << REG_CTRL_TX_DMA_EN_OFFSET)
+#define REG_CTRL_RX_THRES_MSK (0x1f << REG_CTRL_RX_THRES_OFFSET)
+#define REG_CTRL_TX_THRES_MSK (0x1f << REG_CTRL_TX_THRES_OFFSET)
+
+/* Field mask of SPI status register */
+#define REG_STAT_SPI_ACT_OFFSET (0)
+#define REG_STAT_RX_NUM_OFFSET (8)
+#define REG_STAT_RX_EMPTY_OFFSET (14)
+#define REG_STAT_RX_FULL_OFFSET (15)
+#define REG_STAT_TX_NUM_OFFSET (16)
+#define REG_STAT_TX_EMPTY_OFFSET (22)
+#define REG_STAT_TX_FULL_OFFSET (23)
+
+#define REG_STAT_SPI_ACT_MSK (0x1 << REG_STAT_SPI_ACT_OFFSET)
+#define REG_STAT_RX_NUM_MSK (0x1f << REG_STAT_RX_NUM_OFFSET)
+#define REG_STAT_RX_EMPTY_MSK (0x1 << REG_STAT_RX_EMPTY_OFFSET)
+#define REG_STAT_RX_FULL_MSK (0x1 << REG_STAT_RX_FULL_OFFSET)
+#define REG_STAT_TX_NUM_MSK (0x1f << REG_STAT_TX_NUM_OFFSET)
+#define REG_STAT_TX_EMPTY_MSK (0x1 << REG_STAT_TX_EMPTY_OFFSET)
+#define REG_STAT_TX_FULL_MSK (0x1 << REG_STAT_TX_FULL_OFFSET)
```

Comment:
Use `GENMASK` when possible.

Reply:
That's great!
Thanks for your suggestion.